Envelopes/saull_Peak Envelope Generator.lua

```diff
@@ -0,0 +1,277 @@
+invertPrev = false
+offset = 0.0
+offsetPrev = 0.0
+scaling = 1.0
+scalingPrev = 1.0
+peakrate = 10
+peakratePrev = 10
+autoGenerate = true
+autoScale = true
+selectedItems  = {}
+numchannels = 1
+extratype = 0 -- ???
+min = 0
+max = 1.0
+
+function EnvChange()
+  if env == nil then
+    if prevEnv ~= nil then
+      env = prevEnv
+    else
+      envString = 'None'
+    end
+  end
+     
+  if env ~= nil then
+    if pcall(function() local envName = reaper.GetEnvelopeName(env) end) then
+      if env ~= prevEnv then 
+        retval, envString = reaper.GetEnvelopeName(env)
+        SetParamMinMaxCenter(env)
+        envChanged = true
+      end
+      prevEnv = env
+    else
+      env = nil
+      envString = 'None'
+    end
+  end
+end
+
+function SaveSelectedItems()
+  local t = t or {}
+  for i = 0, reaper.CountSelectedMediaItems(0)-1 do
+    t[i+1] = reaper.GetSelectedMediaItem(0, i)
+  end
+  return t
+end
+ 
+function CalculateEnvelope()
+    
+    local lpeakrate = 0
+    
+    if smooth then
+      lpeakrate = peakrate*2 
+    else
+      lpeakrate = peakrate
+    end
+
+    for i, item in ipairs(selectedItems) do
+        local peaks = {}
+        local len = reaper.GetMediaItemInfo_Value(item, "D_LENGTH")
+        local starttime = reaper.GetMediaItemInfo_Value(item, "D_POSITION")
+        local numsamples = math.floor(len*lpeakrate + 0.5)
+        local ticklen = len/numsamples
+        local buf = reaper.new_array(2*numsamples)
+        buf.clear()
+        local retval = reaper.GetMediaItemTake_Peaks(reaper.GetActiveTake(item), lpeakrate, starttime, numchannels, numsamples, extratype, buf)
+        local samplecount  = (retval & 0xfffff)
+        local outmode = (retval & 0xf00000)>>20
+        if samplecount > 0  then
+            for k = 1, numsamples, 1 do
+              local p = #peaks
+              peaks[p+1] = buf[k]
+              peaks[p+2] = buf[numsamples+k]
+            end
+        end
+        
+        reaper.DeleteEnvelopePointRange(env, starttime, starttime + len)
+        
+      --  reaper.ShowConsoleMsg('maxpeak ' .. maxPeak .. '\n' .. 'max Y ' .. envMaxY .. '\n' .. 'scaling ' .. scaling .. '\n')
+        local scaleFunc;
+        if invert then
+          scaleFunc = function (x) return max - (scaling * math.abs(x)) +offset end
+        else
+          scaleFunc = function (x) return scaling * math.abs(x) + offset end
+        end
+      
+        for k, peak in ipairs(peaks) do
+           reaper.InsertEnvelopePoint(env, starttime + (k-1)*ticklen*.5, scaleFunc(peak), 0,  0, true) 
+        end
+        
+        if smooth then reaper.Main_OnCommand(42208,0) end
+    end
+ end
+  
+  
+ -- INIT
+ function Apply()
+    countSelectedItems = reaper.CountSelectedMediaItems(0)
+    reaper.PreventUIRefresh(1) 
+    reaper.Undo_BeginBlock()
+    if smooth then reaper.Main_OnCommand(40331,0) end
+    selectedItems = SaveSelectedItems()
+    CalculateEnvelope()
+    reaper.Undo_EndBlock("peakEnvelope", -1)
```

Comment:
This string is the undo point name shown to the user; wouldn't it be better if it matched the script's name "Peak envelope generator"?

```suggestion
    reaper.Undo_EndBlock("Peak envelope generator", -1)
```